tiled/client/register.py

```diff
@@ -158,40 +158,43 @@ async def register(
     parsed_walkers.extend(DEFAULT_WALKERS)
     prefix_parts = [segment for segment in prefix.split("/") if segment]
     for segment in prefix_parts:
-        child_catalog = await catalog.lookup_adapter([segment])
-        if child_catalog is None:
+        child_node = await anyio.to_thread.run_sync(node.get, segment)
+        if child_node is None:
             key = key_from_filename(segment)
-            await create_node_safe(
-                catalog,
+            await create_node_or_drop_collision(
+                node,
                 structure_family=StructureFamily.container,
+                data_sources=[],
                 metadata={},
                 specs=[],
                 key=key,
             )
-            child_catalog = await catalog.lookup_adapter([segment])
-        catalog = child_catalog
+            # TODO When we have a tiled AsyncClient, use that.
+            child_node = await anyio.to_thread.run_sync(node.get, segment)
+        node = child_node
     if path.is_dir():
         # Recursively enter the directory and any subdirectories.
         if overwrite:
```

Comment:
I have a question about this...when I was doing this on the server side before this PR, I couldn't figure out why you would ever want to to set this to true. It doesn't effec the node you're writing, but it deletes collection parents of the node. Am i missing something? Is there a use case for that?

Reply:
It's meant to delete the node and all its children. The use case is when you walk a directory and then walk it again later, wanting to replace whatever Tiled had from directory with new nodes representing the current (maybe changed) contents.